runtime/src/accounts_index.rs

```diff
@@ -1194,12 +1195,21 @@ impl<T: 'static + Clone + IsCached + ZeroLamport> AccountsIndex<T> {
                     max_clean_root,
                     account_indexes,
                 );
+                empty_key = slot_list.is_empty();
             });
         }
+        if empty_key {
+            let mut w_maps = self.account_maps.write().unwrap();
+            if let Some(x) = w_maps.get(pubkey) {
+                if x.slot_list.read().unwrap().is_empty() {
```

Comment:
As a good exercise, could we add a comment about why this additional check is necessary? What race are we trying to prevent?

Reply:
Done!